readthedocs/projects/tasks.py

```diff
@@ -127,6 +128,10 @@ def run(self, pk, version_pk=None, build_pk=None, record=True, docker=False,
 
             self.config = load_yaml_config(version=self.version)
 
+        if self.setup_env.failed:
+            self.send_notifications()
+            raise BuildEnvironmentError(_('Build Setup Failed'))
```

Comment:
Not sure if it's better here to raise an exception, or just return from the build.